tests/test_wasm_builtin_simd.c

```diff
@@ -0,0 +1,488 @@
+// }
+// i8x16 TESTFN i8x16_shr_s(i8x16 vec, int32_t shift) {
+//   return vec >> shift;
+// }
+// i8x16 TESTFN i8x16_shr_u(i8x16 vec, int32_t shift) {
+//   return (i8x16)((u8x16)vec >> shift);
+// }
+i8x16 TESTFN i8x16_add(i8x16 x, i8x16 y) {
+  return x + y;
+}
+i8x16 TESTFN i8x16_add_saturate_s(i8x16 x, i8x16 y) {
+  return __builtin_wasm_add_saturate_s_i8x16(x, y);
+}
+i8x16 TESTFN i8x16_add_saturate_u(i8x16 x, i8x16 y) {
+  return __builtin_wasm_add_saturate_u_i8x16(x, y);
+}
+i8x16 TESTFN i8x16_sub(i8x16 x, i8x16 y) {
+  return x - y;
+}
+i8x16 TESTFN i8x16_sub_saturate_s(i8x16 x, i8x16 y) {
+  return __builtin_wasm_sub_saturate_s_i8x16(x, y);
+}
+i8x16 TESTFN i8x16_sub_saturate_u(i8x16 x, i8x16 y) {
+  return __builtin_wasm_sub_saturate_u_i8x16(x, y);
+}
+i8x16 TESTFN i8x16_mul(i8x16 x, i8x16 y) {
+  return x * y;
+}
+i16x8 TESTFN i16x8_neg(i16x8 vec) {
+  return -vec;
+}
+int32_t TESTFN i16x8_any_true(i16x8 vec) {
+  return __builtin_wasm_any_true_i16x8(vec);
+}
+int32_t TESTFN i16x8_all_true(i16x8 vec) {
+  return __builtin_wasm_all_true_i16x8(vec);
+}
+// i16x8 TESTFN i16x8_shl(i16x8 vec, int32_t shift) {
+//   return vec << shift;
+// }
+// i16x8 TESTFN i16x8_shr_s(i16x8 vec, int32_t shift) {
+//   return vec >> shift;
+// }
+// i16x8 TESTFN i16x8_shr_u(i16x8 vec, int32_t shift) {
+//   return (i16x8)((u16x8)vec >> shift);
+// }
+i16x8 TESTFN i16x8_add(i16x8 x, i16x8 y) {
+  return x + y;
+}
+// i16x8 TESTFN i16x8_add_saturate_s(i16x8 x, i16x8 y) {
+//   return __builtin_wasm_add_saturate_s_i16x8(x, y);
+// }
+// i16x8 TESTFN i16x8_add_saturate_u(i16x8 x, i16x8 y) {
+//   return __builtin_wasm_add_saturate_u_i16x8(x, y);
+// }
+i16x8 TESTFN i16x8_sub(i16x8 x, i16x8 y) {
+  return x - y;
+}
+i16x8 TESTFN i16x8_sub_saturate_s(i16x8 x, i16x8 y) {
+  return __builtin_wasm_sub_saturate_s_i16x8(x, y);
+}
+i16x8 TESTFN i16x8_sub_saturate_u(i16x8 x, i16x8 y) {
+  return __builtin_wasm_sub_saturate_u_i16x8(x, y);
+}
+i16x8 TESTFN i16x8_mul(i16x8 x, i16x8 y) {
+  return x * y;
+}
+i32x4 TESTFN i32x4_neg(i32x4 vec) {
+  return -vec;
+}
+int32_t TESTFN i32x4_any_true(i32x4 vec) {
+  return __builtin_wasm_any_true_i32x4(vec);
+}
+int32_t TESTFN i32x4_all_true(i32x4 vec) {
+  return __builtin_wasm_all_true_i32x4(vec);
+}
+// i32x4 TESTFN i32x4_shl(i32x4 vec, int32_t shift) {
+//   return vec << shift;
+// }
+// i32x4 TESTFN i32x4_shr_s(i32x4 vec, int32_t shift) {
+//   return vec >> shift;
+// }
+// i32x4 TESTFN i32x4_shr_u(i32x4 vec, int32_t shift) {
+//   return (i32x4)((u32x4)vec >> shift);
+// }
+i32x4 TESTFN i32x4_add(i32x4 x, i32x4 y) {
+  return x + y;
+}
+i32x4 TESTFN i32x4_sub(i32x4 x, i32x4 y) {
+  return x - y;
+}
+i32x4 TESTFN i32x4_mul(i32x4 x, i32x4 y) {
+  return x * y;
+}
+i64x2 TESTFN i64x2_neg(i64x2 vec) {
+  return -vec;
+}
+// int32_t TESTFN i64x2_any_true(i64x2 vec) {
+//   return __builtin_wasm_any_true_i64x2(vec);
+// }
+// int32_t TESTFN i64x2_all_true(i64x2 vec) {
+//   return __builtin_wasm_all_true_i64x2(vec);
+// }
```

Comment:
LLVM doesn't know how to split the operands of these ops up when i64x2 ops are disabled.